From 60f7580959ce34fa0a85ef59b0d76be829d10bb3 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 19 Mar 2017 19:25:34 -0700
Subject: [PATCH] Link libexecinfo if platform provides it

Platforms e.g. FreeBSD do not have backtrace APIs which
are glibc specific, also systems based on musl/linux
are in same boat. This patch checks for libexecinfo
which provides backtrace functionality on these platforms

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 WebBridge/CMakeLists.txt | 3 +++
 WebBridge/WebBridge.cpp  | 4 ++++
 2 files changed, 7 insertions(+)

Index: git/WebBridge/CMakeLists.txt
===================================================================
--- git.orig/WebBridge/CMakeLists.txt
+++ git/WebBridge/CMakeLists.txt
@@ -18,6 +18,8 @@ set(WEBBRIDGE_POLICY 0 CACHE STRING "NA"
 set(WEBBRIDGE_OOMADJUST 0 CACHE STRING "Adapt the OOM score [-15 - 15]")
 set(WEBBRIDGE_STACKSIZE 0 CACHE STRING "Default stack size per thread")
 
+find_library(LIBEXECINFO_LIBRARY NAMES execinfo)
+mark_as_advanced(LIBEXECINFO_LIBRARY)
 
 set(WEBBRIDGE_WEBBRIDGE_LIBS 
     webbridgesupport
@@ -33,7 +35,7 @@ set(WEBBRIDGE_WEBBRIDGE_SOURCES
 
 # add the library
 add_executable(webbridge ${WEBBRIDGE_WEBBRIDGE_SOURCES})
-target_link_libraries(webbridge ${WEBBRIDGE_WEBBRIDGE_LIBS})
+target_link_libraries(webbridge ${WEBBRIDGE_WEBBRIDGE_LIBS} ${LIBEXECINFO_LIBRARY})
 
 install(TARGETS webbridge DESTINATION ${CMAKE_INSTALL_PREFIX}/bin)
 
Index: git/WebBridge/WebBridge.cpp
===================================================================
--- git.orig/WebBridge/WebBridge.cpp
+++ git/WebBridge/WebBridge.cpp
@@ -4,6 +4,10 @@
 #include <syslog.h>
 #endif
 
+#ifdef __linux__
+#include <execinfo.h>
+#endif
+
 MODULE_NAME_DECLARATION(BUILDREF_WEBBRIDGE)
 
 #define MAX_EXTERNAL_WAITS 2000 /* Wait for 2 Seconds */
